guides/creating-a-custom-stepper-using-the-cdk-stepper.md

```diff
@@ -0,0 +1,153 @@
+  <ng-container [ngTemplateOutlet]="step.content"></ng-container>
+  
+  <!-- These buttons are used to navigate between the steps -->
+  <button cdkStepperPrevious>Previous</button>
+  <button cdkStepperNext>Next</button>
+</section>
+```
+
+In the `app.component.scss` file you can now style the stepper however you want. 
+
+## Using our new custom stepper component
+
+Now we are ready to use our new custom stepper component and fill it with steps. Therefore we can, for example, add it to our `app.component.html` and define some steps:
+
+__app.component.html__
+```html
+<app-custom-stepper>
+  <cdk-step>
+    <ng-template cdkStepLabel>Step1</ng-template>
+    <input type="text" name="a" value="a" />
+  </cdk-step>
+  <cdk-step>
+    <ng-template cdkStepLabel>Step2</ng-template>
+    <input type="text" name="b" value="b" />
+  </cdk-step>
+</app-custom-stepper>
+```
+
+As you can see in this example, each step needs to be wrapped inside a `<cdk-step>` tag and you can define a step label using ` <ng-template cdkStepLabel>This is your step label text</ng-template>`.
+
+If you want to iterate over your steps and use your own custom component you can do it, for example, this way:
+
+```typescript
+<app-custom-stepper>
+  <cdk-step
+    *ngFor="let step of mySteps; let stepIndex = index">
+    <ng-template cdkStepLabel>{{ step.title }}</ng-template>
+    <my-step-component [step]="step"></my-step-component>
+  </cdk-step>
+</app-custom-stepper>
+```
+
+## Linear mode
+
+The above example allowed the user to freely navigate between all steps. The `CdkStepper` additionally provides the linear mode which requires the user to complete previous steps before proceeding.
+
+You can either use a single form for the entire stepper or a form for each individual step to validate if a step is completed. Alternatively, you can pass the completed property to each of the steps and set the property value depending on your own logic without using a form.
+
+A simple example without using forms could look this way:
+
+__app.component.html__
+```html
+<app-custom-stepper linear>
+  <cdk-step editable="false" [completed]="completed">
+    <ng-template cdkStepLabel>StepB1</ng-template>
+    <input type="text" name="a" value="Cannot proceed to next step" />
+    <button (click)="completeStep()">Complete step</button>
+  </cdk-step>
+  <cdk-step editable="false">
+    <ng-template cdkStepLabel>StepB2</ng-template>
+    <input type="text" name="b" value="b" />
+  </cdk-step>
+</app-custom-stepper>
+```
+
+__app.component.ts__
+```typescript
+export class AppComponent {
+  completed = false;
+
+  completeStep(): void {
+    this.completed = true;
+  }
+}
+```
+
+The steps are marked as `editable="false"` which means that the user cannot return to this step once it has been marked as completed. It is not possible to navigate to the second step until the first one has been completed by clicking the *Complete Step* button.
```

Comment:
I'm not sure whether we should re-explain things that are already in the API docs.

Reply:
Agree, removed it